components/runners/ambient-runner/ambient_runner/tools/vertex_client.py

```
@@ -0,0 +1,131 @@
+        if tools:
+            payload["tools"] = tools
+
+        req = urllib.request.Request(
+            url,
+            data=json.dumps(payload).encode(),
+            headers={
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {token}",
+            },
+            method="POST",
+        )
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)